libvast/src/system/index.cpp

```diff
@@ -1607,77 +1664,26 @@ index(index_actor::stateful_pointer<index_state> self,
-                      .then(
-                        [self, partition_id, store_path](atom::done) {
-                          VAST_DEBUG("{} erased partition store {} from "
-                                     "filesystem",
-                                     *self, partition_id);
-                        },
-                        [self, partition_id,
-                         store_path](const caf::error& err) {
-                          if (err == ec::no_such_file)
-                            VAST_DEBUG("{} failed to erase store {} at {}: {} ",
-                                       *self, partition_id, store_path, err);
-                          else
-                            VAST_WARN("{} failed to erase store {} at {}: {}",
-                                      *self, partition_id, store_path, err);
-                        });
+                    erase_dense_index_file();
                     rp.deliver(caf::make_error(ec::filesystem_error,
                                                "aborting erasure due to "
                                                "encountering a legacy "
                                                "oversized partition"));
-                  } else {
-                    // TODO: We could send `all_ids` as the second argument
-                    // here, which doesn't really make sense from an interface
-                    // perspective but would save the partition from recomputing
-                    // the same bitmap.
-                    rp.delegate(partition_actor, atom::erase_v);
+                    return;
                   }
-                  // Note that mmap's will increase the reference count of a
-                  // file, so unlinking should not affect indexers that are
-                  // currently loaded and answering a query.
-                  self
-                    ->request<caf::message_priority::high>(
-                      self->state.filesystem, caf::infinite, atom::erase_v,
-                      synopsis_path)
-                    .then(
-                      [self, partition_id](atom::done) {
-                        VAST_DEBUG("{} erased partition synopsis {} from "
-                                   "filesystem",
-                                   *self, partition_id);
-                      },
-                      [self, partition_id,
-                       synopsis_path](const caf::error& err) {
-                        VAST_WARN("{} failed to erase partition "
-                                  "synopsis {} at {}: {}",
-                                  *self, partition_id, synopsis_path, err);
-                      });
+                  // TODO: We could send `all_ids` as the second
+                  // argument here, which doesn't really make sense
+                  // from an interface perspective but would save the
+                  // partition from recomputing the same bitmap.
+                  auto partition_actor
```

Comment:
You probably still want to explicitly drop this in case we don't get to the fallback path, otherwise there's a deleted partition hanging around in the cache.

Reply:
I don't think that would be a problem.

Reply:
Tobias and I talked about this and agree that this is not a problem; the partition will be rotated out of the cache.